src/js/index.d.ts

```diff
@@ -39,7 +39,7 @@ declare module "lottie-react-native" {
      * animation will correspondingly update to the frame at that progress value. This
      * prop is not required if you are using the imperative API.
      */
-    progress?: number | Animated.Value;
+    progress?: number | Animated.Value | any;
```

Comment:
perhaps do the same as the try import on the js file,
type AnimatedSource = Animated.Value
try {
 Reanimated = require(...).default
AnimatedSource = Animated.Value | Reanimated.Node
}

dont know if is gonna work though, I'm seen all of this today after my MR haha

Reply:
turns out you can't reassign a type, but since the class implementation is so basic (https://github.com/software-mansion/react-native-reanimated/blob/master/react-native-reanimated.d.ts) it could be defined within the d.ts file to be equal to the original one, since typescript only cares about the same implementation, and not which class is which.
something like:

interface ReanimatedNode {
 new (nodeConfig: object, inputNodes?: ReadonlyArray<AnimatedNode>): ReanimatedNode
isNativelyInitialized(): boolean
' __value': any
}

should suffice